Spec/RealtimeClientConnection.swift

```diff
@@ -2093,6 +2101,73 @@ class RealtimeClientConnection: QuickSpec {
                         expect(channel.errorReason).to(beIdenticalTo(protocolError.error))
                     }
 
+                    it("should resume the connection after an auth renewal and detach all channels if the connectionId changes") {
+                        let options = AblyTests.commonAppSetup()
+                        options.disconnectedRetryTimeout = 1.0
+                        options.tokenDetails = getTestTokenDetails(ttl: 2.0)
+                        let client = AblyTests.newRealtime(options)
+                        defer { client.dispose(); client.close() }
+
+                        let channel = client.channels.get("test")
+                        waitUntil(timeout: testTimeout) { done in
+                            channel.attach() { error in
+                                expect(error).to(beNil())
+                                done()
+                            }
+                        }
+
+                        let initialConnectionId = client.connection.id
+
+                        guard let firstTransport = client.transport as? TestProxyTransport else {
+                            fail("TestProxyTransport is not set"); return
+                        }
+
+                        // Wait for token to expire
+                        waitUntil(timeout: testTimeout) { done in
+                            let partialDone = AblyTests.splitDone(2, done: done)
+                            channel.once(.Detached) { error in
```

Comment:
Why is the channel becoming detached?  If a connection is closed because of an auth failure, the channel does not become detached, it remains attached until the connection is closed or moves to the suspended state?